proxy/src/control/destination/mod.rs

```diff
@@ -144,6 +160,12 @@ impl<B> Resolution<B> {
     }
 }
 
+impl<B> Drop for Resolution<B> {
+    fn drop(&mut self) {
+        self.receiver_gone.store(true, Ordering::Release);
+    }
+}
```

Comment:
This is probably correct, but using `Arc<()>` and `Weak<()>` would automate away the need for this.

Reply:
thanks, that seems like a much nicer solution